File: abc064/src/c.cpp
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;

int main()
{
    int n;
    cin >> n;
    vector<int> a(n); // rate
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }
    const int colors = 8; // 7 categories + 1 category for rates 3200 and above
    vector<int> counts(colors + 1, 0);
    for (const int& a_i : a) {         // iterate over the rates
        int index = min(a_i / 400, 8); // determine the category index
        counts[index]++;               // increment the count for the corresponding category
    }
    int count = 0;
    for (int i = 0; i < colors; i++) {
        if (counts[i] > 0) {
            count++; // increment the count of distinct colors
        }
    }
    int min = max(count, counts[8] > 0 ? 1 : 0); // calculate the minimum number of distinct colors
    int max = count + counts[8];                 // calculate the maximum number of distinct colors
    cout << min << " " << max << endl;

    return 0;
}

File: abc064/src/bin/c.rs
use proconio::input;

fn main() {
    input! {
        n: usize,
        a: [usize; n],
    };
    let colors = 8; //7種類+3200以上の1種類
    let mut counts = vec![0_usize; colors + 1];
    for a_i in a.into_iter().map(|a_i| a_i.min(3200)) {
        //a_iはレーティング
        counts[a_i / 400] += 1; //3200以上の人が何人か数える
    }
    let mut count = 0_usize;
    for c in counts.iter().copied().take(colors) {
        if c > 0 {
            count += 1;
        }
    }
    let min = count.max(if counts[colors] > 0 { 1 } else { 0 });
    let max = count + counts[colors];
    println!("{} {}", min, max);
}
